### PR TITLE
added authentication and toast component

### PR DESCRIPTION
**Pull request by ashishnnnnn**
-> Added Authentication 
-> Added Basic Validation on input
-> Added toast component

### CODE REVIEW

File: src/Pages/Signup/Signup.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          email: email,
+          password: password,
+        });
+        localStorage.setItem("token", response.data.encodedToken);
+        setAuthState({ type: "TOKEN", payload: response.data.encodedToken });
+        handleaddtoast({
+          message: `Welcome ${firstName}`,
+          type: "alert-success",
+        });
+        navigate("/");
```
Comment (lannisterrr) [inline src/Pages/Signup/Signup.js:37]: use the "from" state to naviagate the user to the clicked page.

File: src/index.js [hunk 2]
```diff
@@ -6,19 +6,25 @@ import { makeServer } from "./server";
...
 ReactDOM.render(
   <React.StrictMode>
-    <Router>
-      <FilterProvider>
-        <VideoListProvider>
-          <App />
-        </VideoListProvider>
-      </FilterProvider>
-    </Router>
+    <AuthProvider>
```
Comment (lannisterrr) [inline src/index.js]: Router should be the top most element as you might need to use it.

**Review by lannisterrr — COMMENTED** [review]
overall clean code some suggesstions made.

**Review by nerdyspook — APPROVED** [review]
Well structured code, easy to read.

**Review by nerdyspook — COMMENTED** [review]
Please do look into the URLs you have provided for image src since the links seems to be broken.

mohit

**Review by Manpreet-hub — APPROVED** [review]
Code looks clean and Well structured.

File: src/index.js (hunk 2 above)
Comment (ashishnnnnn) [reply src/index.js]: Done..

File: src/Pages/Signup/Signup.js (hunk 1 above)
Comment (ashishnnnnn) [reply src/Pages/Signup/Signup.js:37]: Okay..Will Use it..

**Pull request merged**